### PR TITLE
[ROCm] enable additional missed cpp tests (related to test_api binary)

### PR DESCRIPTION
**Pull request by hongxiayang**
[ROCm] enable additional missed cpp tests (related to test_api binary)
InitTest/ParallelTest

cc @jeffdaily @sunway513 @jithunnair-amd @pruthvistony @ROCmSupport @dllehr-amd @jataylo @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10 @gujinghui @PenghuiCheng @jianyuh @min-jean-cho @yanbing-j @Guobing-Chen @Xia-Weiwen @mcarilli @ptrblck @leslie-fang-intel @EikanWang @voznesenskym @penguinwu @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov @ColinPeppler @ngimel

### CODE REVIEW

File: test/cpp/tensorexpr/CMakeLists.txt [hunk 1]
```diff
@@ -26,7 +26,7 @@ set(TENSOREXPR_TEST_SRCS
   ${TENSOREXPR_TEST_ROOT}/test_type_specializations.cpp
 )
 
-if(USE_CUDA)
+if(USE_CUDA || USE_ROCM)
```
Comment (jeffdaily) [inline test/cpp/tensorexpr/CMakeLists.txt:29]:
```suggestion
if(USE_CUDA OR USE_ROCM)
```

File: test/cpp/api/CMakeLists.txt [hunk 2]
```diff
@@ -40,12 +41,11 @@ set(TORCH_API_TEST_SOURCES
   ${TORCH_API_TEST_DIR}/grad_mode.cpp
   ${TORCH_API_TEST_DIR}/operations.cpp
 )
-if(USE_CUDA)
+if(USE_CUDA || USE_ROCM)
```
Comment (jeffdaily) [inline test/cpp/api/CMakeLists.txt:44]:
```suggestion
if(USE_CUDA OR USE_ROCM)
```

**Comment by hongxiayang** [comment]
Enabled: test_api::ParallelTest::DataParallelUsesAllAvailableCUDADevices_CUDA in 1 of 3: https://pipelines.actions.githubusercontent.com/serviceHosts/7d146c05-69c3-4c20-a0e7-818111670117/_apis/pipelines/1/runs/3269189/signedlogcontent/208?urlExpires=2023-09-07T20%3A07%3A25.0988511Z&urlSigningMethod=HMACV1&urlSignature=nrVxzYwASyX0orGO7EaQ4t4JKuHYWh9D6BaaU6va%2BpM%3D

```
2023-08-18T05:54:24.7230779Z ../../../../../opt/conda/envs/py_3.8/lib/python3.8/site-packages/torch/bin/test_api::ParallelTest::DataParallelUsesAllAvailableCUDADevices_CUDA 
2023-08-18T05:54:24.7231737Z [gw3] [ 99%] PASSED [1.0435s] ../../../../../opt/conda/envs/py_3.8/lib/python3.8/site-packages/torch/bin/test_api::GradModeTest::TestRequiresGradViewOpExiting 
2023-08-18T05:54:24.7232544Z ../../../../../opt/conda/envs/py_3.8/lib/python3.8/site-packages/torch/bin/test_api::OperationTest::Linear_out 
2023-08-18T05:54:24.7233443Z [gw2] [ 99%] PASSED [0.9581s] ../../../../../opt/conda/envs/py_3.8/lib/python3.8/site-packages/torch/bin/test_api::OperationTest::Lerp 
2023-08-18T05:54:24.7234271Z [gw0] [ 99%] PASSED [1.1766s] ../../../../../opt/conda/envs/py_3.8/lib/python3.8/site-packages/torch/bin/test_api::OperationTest::Cross 
2023-08-18T05:54:24.7235083Z [gw3] [ 99%] PASSED [1.0419s] ../../../../../opt/conda/envs/py_3.8/lib/python3.8/site-packages/torch/bin/test_api::OperationTest::Linear_out 
2023-08-18T05:54:24.7236053Z [gw1] [100%] PASSED [1.2059s] ../../../../../opt/conda/envs/py_3.8/lib/python3.8/site-packages/torch/bin/test_api::ParallelTest::DataParallelUsesAllAvailableCUDADevices_CUDA 
2023-08-18T05:54:24.7236479Z 
```

InitTest:
```
2023-08-18T05:54:24.6457624Z ../../../../../opt/conda/envs/py_3.8/lib/python3.8/site-packages/torch/bin/test_api::InitTest::CalculateGainWithLeakyRelu 
2023-08-18T05:54:24.6458128Z [gw3] [ 53%] PASSED [0.9833s] ../../../../../opt/conda/envs/py_3.8/lib/python3.8/site-packages/torch/bin/test_api::ModulesTest::Tanh 
2023-08-18T05:54:24.6458569Z ../../../../../opt/conda/envs/py_3.8/lib/python3.8/site-packages/torch/bin/test_api::ModulesTest::Tanhshrink 
2023-08-18T05:54:24.6459102Z [gw2] [ 53%] PASSED [0.9635s] ../../../../../opt/conda/envs/py_3.8/lib/python3.8/site-packages/torch/bin/test_api::ModuleListTest::RangeBasedForLoop 
2023-08-18T05:54:24.6459539Z ../../../../../opt/conda/envs/py_3.8/lib/python3.8/site-packages/torch/bin/test_api::ModuleListTest::InvalidAt 
2023-08-18T05:54:24.6460081Z [gw0] [ 53%] PASSED [0.9349s] ../../../../../opt/conda/envs/py_3.8/lib/python3.8/site-packages/torch/bin/test_api::ModulesTest::BatchNorm1dStateful 
2023-08-18T05:54:24.6460541Z ../../../../../opt/conda/envs/py_3.8/lib/python3.8/site-packages/torch/bin/test_api::ModulesTest::BatchNorm1dStateless 
2023-08-18T05:54:24.6461095Z [gw1] [ 53%] PASSED [0.9001s] ../../../../../opt/conda/envs/py_3.8/lib/python3.8/site-packages/torch/bin/test_api::InitTest::CalculateGainWithLeakyRelu 
2023-08-18T05:54:24.6461587Z ../../../../../opt/conda/envs/py_3.8/lib/python3.8/site-packages/torch/bin/test_api::InitTest::CanInitializeCnnWithOrthogonal 
```

**Comment by hongxiayang** [comment]
3 ROCM tests were successful
image

**Comment by hongxiayang** [comment]
other ci failures were not related to this change.

**Pull request closed without merging**